test/unit/inbound/process-submitted-settlement.test.js

```diff
@@ -0,0 +1,137 @@
+const mockCommit = jest.fn()
+const mockRollback = jest.fn()
+const mockTransaction = {
+  commit: mockCommit,
+  rollback: mockRollback
+}
+
+jest.mock('../../../app/data', () => {
+  return {
+    sequelize:
+       {
+         transaction: jest.fn().mockImplementation(() => {
+           return { ...mockTransaction }
+         })
+       }
+  }
+})
+
+jest.mock('../../../app/inbound/get-settlement-by-invoicenumber-and-value')
+const getSettlementByInvoiceNumberAndValue = require('../../../app/inbound/get-settlement-by-invoicenumber-and-value')
+
+jest.mock('../../../app/inbound/save-settlement')
+const saveSettlement = require('../../../app/inbound/save-settlement')
+
+const processReturnSettlement = require('../../../app/inbound/process-return-settlement')
+const settlement = JSON.parse(JSON.stringify(require('../../mock-settlement')))
+
+describe('process return settlement request', () => {
+  beforeEach(() => {
+    getSettlementByInvoiceNumberAndValue.mockResolvedValue(undefined)
+    saveSettlement.mockResolvedValue(undefined)
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should call getSettlementByInvoiceNumberAndValue when a valid settlement is given', async () => {
+    await processReturnSettlement(settlement)
+    expect(getSettlementByInvoiceNumberAndValue).toBeCalled()
+  })
+
+  test('should call getSettlementByInvoiceNumberAndValue once when a valid settlement is given', async () => {
+    await processReturnSettlement(settlement)
+    expect(getSettlementByInvoiceNumberAndValue).toBeCalledTimes(1)
+  })
+
+  test('should call getSettlementByInvoiceNumberAndValue with "settlement" and "mockTransaction" when valid settlement is given', async () => {
+    await processReturnSettlement(settlement)
+    expect(getSettlementByInvoiceNumberAndValue).toHaveBeenCalledWith(settlement.invoiceNumber, settlement.value, mockTransaction)
+  })
+
+  test('should throw when getSettlementByInvoiceNumberAndValue throws', async () => {
+    getSettlementByInvoiceNumberAndValue.mockRejectedValue(new Error('Database retrieval issue'))
+    const wrapper = async () => {
+      await processReturnSettlement(settlement)
+    }
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw error when getSettlementByInvoiceNumberAndValue throws', async () => {
```

Comment:
would capitalise "Error" in the test name (and for the other sim. tests) as we are testing that the correct Error class is thrown -- if we create our our DatabaseRetrievalError, then this test would become, "should throw DatabaseRetrievalError when getSettlementByInvoiceNumberAndValue throws"

Reply:
Line 122 is the same